src/main/java-templates/org/postgresql/Driver.java

```diff
@@ -411,7 +411,7 @@ public Connection getResult(long timeout) throws SQLException {
      * @throws SQLException if the connection could not be made
      */
     private static Connection makeConnection(String url, Properties props) throws SQLException {
-        return new @JDBCCONNECTCLASS@(hostSpecs(props),
+        return new ${JDBCCONNECTCLASS}(hostSpecs(props),
```

Comment:
@vlsi Presumably this pre-processing step was added for performance otherwise reflection would have been used. 

We could simply have a properties file that has the same templating step which could contain the spec version the driver is. Then we could return the appropriate instance, which at the moment is the JDBC4, 4.1 or 4.2 implementation.

@mkarg Change is good, but we should do the best we can not to break clients. If there is no other reasonable way then a breaking change can be made. It's not a massive change but as @vlsi mentions it will invalidate all the existing documentation out there on external sites.